### PR TITLE
FontFeature diagrams

### PR DESCRIPTION
**Pull request by Hixie**
Adds a bunch of samples for FontFeature.

### CODE REVIEW

**Comment by Hixie** [comment]
See https://github.com/flutter/engine/pull/23730 for what these document.

**Comment by Hixie** [comment]
cc @GaryQian

File: utils/diagram_generator/lib/main.dart [hunk 1]
```diff
@@ -62,6 +62,8 @@ Future<void> main() async {
   final List<String> categories = flags['category'] as List<String>;
   final List<String> names = flags['name'] as List<String>;
 
+  print('Filters:\n  categories: $categories\n  names: $names');
```
Comment (GaryQian) [inline utils/diagram_generator/lib/main.dart:65]: Nit: Was this meant to be for debug or actually included here?

File: utils/diagram_generator/pubspec.yaml [hunk 2]
```diff
@@ -26,10 +26,50 @@ environment:
...
+      fonts:
+        - asset: fonts/SortsMillGoudy-Regular.ttf
+
+    - family: Source Code Pro
+      fonts:
+        - asset: fonts/SourceCodePro-Bold.ttf
+
+    - family: Ubuntu Mono
+      fonts:
+        - asset: fonts/UbuntuMono-Bold.ttf
```
Comment (GaryQian) [inline utils/diagram_generator/pubspec.yaml:75]: Nit trailing newline

**Review by GaryQian — APPROVED** [review]
LGTM, looks really comprehensive!

File: utils/diagram_generator/lib/main.dart (hunk 1 above)
Comment (Hixie) [reply utils/diagram_generator/lib/main.dart:65]: It was intended to stay. The generator is really only run by us, so it's useful to see what it's doing. Previously we didn't report what filters we applied. (This is especially helpful on Linux where we don't currently handle arguments so the filters you specify on the command line don't actually apply and previously that was very confusing.)

File: utils/diagram_generator/pubspec.yaml (hunk 2 above)
Comment (Hixie) [reply utils/diagram_generator/pubspec.yaml:75]: good catch, fixed

**Comment by Hixie** [comment]
Thanks, will land on green.

**Pull request merged**